### PR TITLE
添加无需根证书情况

### PR DESCRIPTION
**Pull request by ZhengQingchen**
部分环境无需根证书情况，证书配置最好是做成可选的.
image

### CODE REVIEW

File: lib/wechat_pay/api/api.ex [hunk 1]
```diff
@@ -117,15 +117,20 @@ defmodule WechatPay.API do
...
-        cert: config.ssl_cert |> decode_public(),
-        key: config.ssl_key |> decode_private()
-      ]
-    ]
-
-    Client.post("secapi/pay/refund", attrs, opts, config)
+    ssl =
+      case config.ssl_cacert do
+        nil ->
+          [cert: config.ssl_cert |> decode_public(), key: config.ssl_key |> decode_private()]
```
Comment (linjunpop) [inline lib/wechat_pay/api/api.ex]: Multiline here would be appreciated. Thanks.

Comment (ZhengQingchen) [reply lib/wechat_pay/api/api.ex]: OK.

**Comment by linjunpop** [comment]
Hi @ZhengQingchen, Thanks very much for contributing. And it would be great to make the CA cert an optional configuration option.

I'm appreciated if we can improve these: https://codebeat.co/projects/github-com-linjunpop-wechat_pay-master/pull_requests/160452

🤝

File: lib/wechat_pay/api/api.ex (hunk 1 above)
Comment (linjunpop) [reply lib/wechat_pay/api/api.ex]: 👍

**Comment by ZhengQingchen** [comment]
I try my best to match the codebeat quality need. It's so hard and inhumane.

**Comment by linjunpop** [comment]
😂

**Comment by linjunpop** [comment]
If that make code too complex, please feel free to rollback to the initial one, and sorry for the inconvenience.

I think easy to understand code is better than a complex but looks like *good* one.

File: lib/wechat_pay/api/api.ex [hunk 2]
```diff
@@ -157,19 +151,25 @@ defmodule WechatPay.API do
...
+
+  defp load_ssl(config) do
+    [
+      cacerts: config.ssl_cacert |> decode_public(),
+      cert: config.ssl_cert |> decode_public(),
+      key: config.ssl_key |> decode_private()
+    ]
+  end
+
+  defp valua_not_nil({_key, value}), do: not is_nil(value)
```
Comment (linjunpop) [inline lib/wechat_pay/api/api.ex]: valua => value

File: lib/wechat_pay/api/api.ex [hunk 3]
```diff
@@ -117,15 +117,9 @@ defmodule WechatPay.API do
...
           {:ok, map} | {:error, WechatPay.Error.t() | HTTPoison.Error.t()}
   def refund(attrs, config) do
-    opts = [
-      ssl: [
-        cacerts: [config.ssl_cacert |> decode_public()],
-        cert: config.ssl_cert |> decode_public(),
-        key: config.ssl_key |> decode_private()
-      ]
-    ]
+    ssl = config |> load_ssl() |> Enum.filter(&valua_not_nil/1)
```
Comment (linjunpop) [inline lib/wechat_pay/api/api.ex]: The tests are broken. Typo here.

**Comment by linjunpop** [comment]
👍

**Comment by linjunpop** [comment]
@ZhengQingchen https://hex.pm/packages/wechat_pay/0.7.0 released.

**Pull request merged**